lib/inspec/input_registry.rb

```diff
@@ -256,7 +256,7 @@ def bind_inputs_from_input_files(profile_name, file_list)
 
         data.inputs.each do |input_name, input_value|
           evt = Inspec::Input::Event.new(
-            value: input_value,
+            value: input_value.is_a?(Hash) ? Thor::CoreExt::HashWithIndifferentAccess.new(input_value) : input_value,
```

Comment:
For consistency, I wonder if we should make a similar change in `bind_inputs_from_runner_api`? Testing that would be difficult, however.

Reply:
Yes for consistency that seems right. Let me look into the testing front for this.

Reply:
Can we add a method to handle this conversion `input_value.is_a?(Hash) ? Thor::CoreExt::HashWithIndifferentAccess.new(input_value) : input_value` so that we can just call that method wherever required and code will be maintainable ?

Reply:
Yes these checks could be re-used by having just a method :)

Reply:
@clintoncwolfe I observed this issue not only additionally happens for runner api input but also for other ways of input. Will try an alternate solution which could work for all the input types, rather than handling it in each method definition for different input types. Hope that would be a better solution.